bridges/centralized-ethereum/src/actors/dr_reporter.rs

```diff
@@ -162,10 +178,23 @@ impl Handler<DrReporterMsg> for DrReporter {
                 return;
             }
 
-            let max_gas_price = get_max_gas_price(&msg, &wrb_contract).await;
+            // TODO: max_gas_price is the same for all batches, it could be calculated per-batch
+            let mut report_gas_price = get_max_gas_price(&msg, &wrb_contract).await;
+            let network_gas_price = match eth.gas_price().await {
+                Ok(x) => x,
+                Err(e) => {
+                    log::error!("Error estimating network gas price: {}", e);
 
-            if max_gas_price == U256::from(0u8) {
-                // Error reading gas price, abort
+                    return;
+                }
+            };
+            let ratio = u256_div_as_f64(network_gas_price, report_gas_price);
+            if ratio <= report_result_max_network_gas_price_ratio {
+                // If not higher than the allowed ratio, set gas price
+                report_gas_price = network_gas_price;
+            } else {
+                // Higher network gas price: return error and abort
+                log::error!("Network gas price is higher than allowed ratio. Ratio: {}. Network gas price: {}. Requests gas price: {}", ratio, network_gas_price, report_gas_price);
```

Comment:
@guidiaz Are we sure that we want to skip the reporting instead of using `report_gas_price * ratio` ?

Reply:
if we did so the "eth_estimateGas" would possibly fail in some providers ... we could try, but then we should still log a warning message, so we can be noticed when reading the logs what's actually happening

Reply:
Agreed. Let's do that @tmpolaczyk